c7n/resources/rds.py

```diff
@@ -281,6 +282,107 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('upgrade-available')
+class UpgradeAvailable(Filter):
+
+    schema = type_schema('upgrade-available')
+
+    def _list_versions(self, client, resource):
+        """ Scan DB instances for available engine upgrades
+
+            This will pull DB instances & check their specific engine for any
+            engine version with higher release numbers than the current one
+
+            This will also tag the rds instance with 'target_engine' which is
+            the most recent version of the engine available
+        """
+        version = resource['EngineVersion']
+        versions = client.describe_db_engine_versions(
+            Engine=resource['Engine'],
+            EngineVersion=version)['DBEngineVersions'][0]
+
+        for v in versions['ValidUpgradeTarget']:
+            if v['IsMajorVersionUpgrade']:
+                continue
+
+            if parse_version(v['EngineVersion']) > parse_version(version):
+                version = v['EngineVersion']
+        return version
+
+    def _tag_rds(self, client, resource, version):
+        client.add_tags_to_resource(
+            ResourceName=resource['DBInstanceArn'],
+            Tags=[{
+                'Key': 'engine_target_upgrade',
+                'Value': version}])
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+        results = []
+        engines = {}
+        for r in resources:
+            if r['Engine'] not in engines:
+                engines[r['Engine']] = self._list_versions(client, r)
+
+            if r['EngineVersion'] == engines[r['Engine']]:
+                continue
+            if parse_version(
+                    engines[r['Engine']]) > parse_version(r['EngineVersion']):
+                if not r in results:
+                    self._tag_rds(client, r, engines[r['Engine']])
```

Comment:
hmm.. by annotation from previous review, i actually meant on the resource dictionary, not tag on the underlying resource. ie. r['c7n.rds-minor-engine-upgrade'] = upgrade_version 
annotations will get serialized out with the resource to s3.